contracts/linear/src/validator_pool.rs

```diff
@@ -340,6 +340,120 @@ impl ValidatorPool {
     }
 }
 
+// V2 Unstake strategy impl
+impl ValidatorPool {
+    /// Strategy
+    /// Step 1. If there is a validator whose `delta` is greater than `total_amount_to_unstake`,
+    ///         it will be selected (If more than one validator match the condition, select one
+    ///         which has smallest `delta`).
+    /// Step 2. If we didn't find any valid validator in step 1, we should select a validator which
+    ///         has largest `delta / target_amount` (If `target_amount` is 0, select one which
+    ///         has largest `delta`).
+    pub fn get_candidate_to_unstake_v2(
+        &self,
+        total_amount_to_unstake: Balance,
+        total_staked_near_amount: Balance,
+    ) -> Option<CandidateValidator> {
+        let mut candidate_validators = self.extract_candidate_validators(total_staked_near_amount);
+
+        Self::sort_candidate_validators_by_delta_asc(&mut candidate_validators);
+
+        let candidate = candidate_validators
+            .iter()
+            .find(|(_validator, _target_amount, delta)| *delta >= total_amount_to_unstake);
+
+        if let Some((validator, _target_amount, _delta)) = candidate {
+            return Some(CandidateValidator {
+                validator: validator.clone(),
+                amount: total_amount_to_unstake,
+            });
+        };
+
+        Self::sort_candidate_validators_by_delta_to_target_desc(&mut candidate_validators);
+
+        candidate_validators
+            .first()
+            .map(|(validator, target_amount, _delta)| {
+                let amount_to_unstake = min3(
+                    total_amount_to_unstake,
+                    if *target_amount != 0 {
+                        target_amount / 2
```

Comment:
can we have some comments to explain why unstake amount could be half of target_amount? this is not as intuitive as the other two values

Reply:
This has changed, please view the newest commit.